### DISCUSSION TITLE
module 'PIL.Image' has no attribute 'LINEAR' [workaround]

### DISCUSSION THREAD

**Discussion by cibernicola** [vladmandic/sdnext#2599] [Ideas]
Using controlnet (builtin version) + segmentation seg_ufade20k model resulting in: 

```
class ExtentTransform(Transform):
  File ****:\automatic\extensions-builtin\sd-webui-controlnet\annotator\oneformer\detectron2\data\transforms\transform.py", line 46, in ExtentTransform
    def __init__(self, src_rect, output_size, interp=Image.LINEAR, fill=0):
AttributeError: module 'PIL.Image' has no attribute 'LINEAR'
```
Workaround:
replace Image.LINEAR with Image.BILINEAR in line 46 within file automatic\extensions-builtin\sd-webui-controlnet\annotator\oneformer\detectron2\data\transforms\transform.py

https://pillow.readthedocs.io/en/stable/deprecations.html

Constants Deprecated since version 9.1.0.

Removed in version 10.0.0.

A number of constants have been removed. Instead, enum.IntEnum classes have been added.

Additional Image constants were deprecated in Pillow 9.1.0, but that was reversed in Pillow 9.4.0 and those constants will now remain available. 

Removed                  |             Use instead

Image.LINEAR           |             Image.BILINEAR or Image.Resampling.BILINEAR

........

**Comment by vladmandic** [comment]
built-in or not, its still controlnet - report to that repo.

**Reply by cibernicola** [reply]
![image](https://github.com/vladmandic/automatic/assets/4579387/421a1365-6650-4ed1-9eb8-9ee4ec498ca6)
